src/compiler/checker.ts

```diff
@@ -16729,6 +16749,24 @@ namespace ts {
             return true;
         }
 
+        function checkAndAggregateParameterExpressionTypes(parameter: ParameterDeclaration): Type[] {
+            const func = <FunctionLikeDeclaration>parameter.parent
+            const usageTypes: Type[] = []
+            forEachInvocation(<Block>func.body, invocation => {
+                const usages = invocation.arguments
+                    .map((arg, i) => ({ arg, symbol: getSymbolAtLocation(arg), i }))
+                    .filter(({ symbol }) => symbol && symbol.valueDeclaration === parameter)
+                if (!usages.length) return
+                const funcSymbol = getSymbolAtLocation(invocation.expression)

```

Comment:
Need to find robust mechanism for resolving the node corresponding to the invoked function to a function signature. Right now things like `interface Foo{ (bar: Bar): Baz }` can mess it up.